libcontainer/container_linux.go

```diff
@@ -1124,6 +1133,16 @@ func (c *linuxContainer) runType() (Status, error) {
 		}
 		return Stopped, newSystemErrorWithCausef(err, "sending signal 0 to pid %d", pid)
 	}
+	// check if the init process is still the same process as the initial one,
+	// it could happen that the process is exited and a new process created with
+	// the same pid, in this case, the container would already be stopped.
+	startTime, err := system.GetProcessStartTime(pid)
+	if err != nil {
+		return Stopped, newSystemErrorWithCausef(err, "getting init process %d start time", pid)
+	}
+	if c.initProcessStartTime != startTime {
+		return Stopped, nil
```

Comment:
I looked into it, and doing a `stat` is not guaranteed to give you a unique inode number (as far as I can tell, inode numbers in procfs are taken from a pool of unused inode numbers). So I think start time (while ugly) is better. But I agree that we should abstract the actual value.